Formula/stolon.rb

```diff
@@ -0,0 +1,30 @@
+class Stolon < Formula
+  desc "Cloud native PostgreSQL manager for high availability"
+  homepage "https://github.com/sorintlab/stolon"
+  url "https://github.com/sorintlab/stolon.git", :tag => "v0.13.0"
+
+  depends_on "go" => :build
+  depends_on "postgresql"
+
+  def install
+    contents = Dir["{*,.git,.gitignore,.github,.travis.yml}"]
+    (buildpath/"src/github.com/sorintlab/stolon").install contents
+    ENV["GOPATH"] = buildpath
+
+    Dir.chdir buildpath/"src/github.com/sorintlab/stolon" do
+      system "./build"
+      bin.install Dir["bin/*"]
+    end
+  end
+
+  test do
+    run_output = shell_output("#{bin}/stolonctl version 2>&1")
+    assert_match "stolonctl version v0.13.0", run_output
+    run_output = shell_output("#{bin}/stolon-keeper --version 2>&1")
+    assert_match "stolon-keeper version v0.13.0", run_output
+    run_output = shell_output("#{bin}/stolon-sentinel --version 2>&1")
+    assert_match "stolon-sentinel version v0.13.0", run_output
+    run_output = shell_output("#{bin}/stolon-proxy --version 2>&1")
+    assert_match "stolon-proxy version v0.13.0", run_output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @fxcoudert Thanks for your inputs. I added test for `stolonctl`(refer my commit). I am still figuring out how to write tests for other three (stolon-keeper,stolon-sentinel,stolon-proxy) as they are long running processes. Is that okay if check only the version for long running processes? Can you please provide some direction. Thanks.

Reply:
I think having a full test for one of them is probably enough, thanks

Reply:
Thanks for your inputs. I squashed the commits and did force push to have clean history.